### PR TITLE
PAPP-25472: Initial commit for the app

### PR DESCRIPTION
**Pull request by hsrivastava-crest**
Please ensure your pull request (PR) adheres to the following guidelines:

- Please refer to our contributing documentation for any questions on submitting a pull request, link: [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md)

#### Please check if your PR fulfills the following requirements:
- [x] Testing of all the changes has been performed (for bug fixes / features)
- [x] The readme.html has been reviewed and added / updated if needed (for bug fixes / features)
- [x] Use the following format for the PR description: `<App Name>: <PR Type> - <PR Description>`
- [x] Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.
- [ ] Verify all checks are passing.
- [x] Do NOT use the `next` branch of the forked repo. Create separate feature branch for raising the PR.
- [x] Do NOT submit updates to dependencies unless it fixes an issue.
#### Please check the type of change your PR introduces:
- [x] New App
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation
- [ ] Other (please describe): 

## Security Considerations (REQUIRED)
- [x] If you are exposing any endpoints using a [REST handler](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers), 
  please document them in the `readme.html`.
- [x] If this is a new connector or you are adding new actions
    - [x] Please document in the `readme.html` all methods (eg, OAuth) used to authenticate 
      with the service that the connector is integrating with.
    - [x] If any actions are unable to run on SOAR Cloud, please document this in the `readme.html`.
- [ ] Are you introducing any new cryptography modules? If yes, please elaborate their purpose:
- [ ] Are you are accessing the file system? If yes, please verify that you are only accessing paths returned through 
the [Vault](https://docs.splunk.com/Documentation/SOARonprem/5.2.1/DevelopApps/AppDevAPIRef#Vault) API.
- [x] Are you are marking code to be ignored by Semgrep with [`nosemgrep`](https://semgrep.dev/docs/ignoring-files-folders-code/#ignoring-code-through-nosemgrep)?
If yes, please provide justification in an additional comment next to the ignored code.

  

## Release Notes (REQUIRED)
- Provide release notes as part of the PR submission which describe high level points about the changes for the upcoming GA release.

## What is the current behavior? (OPTIONAL)
- Describe the current behavior that you are modifying.

## What is the new behavior? (OPTIONAL)
- Describe the behavior or changes that are being added by this PR.

## Other information (OPTIONAL)
- Any other information that is important to this PR such as screenshots of how the component looks before and after the change.

## Pay close attention to (OPTIONAL)
- Any specific code change or test case points which must be addressed/reviewed at the time of GA release.

## Screenshots (if relevant)

---
Thanks for contributing!

### CODE REVIEW

**Comment by hsrivastava-crest** [comment]
I have used "nosemgrep" at one place in the code for the authorization URL. I have tried to remove but the semgrep pipeline is failing https://github.com/splunk-soar-connectors/microsoft365defender/pull/2/commits/e0d2f0f50cea3f84968a057083615336e817b366 
I have also seen that this comment is used for the MSTeams app and WindowsDefenderATP app in the same place.

**Comment by hsrivastava-crest** [comment]
The app integration tests and static test pipelines are failing as the asset and playbook for this app are not merged yet and this is the new app connector. We won't merge this PR until they get merged in the next and the pipeline passes.

**Hence, I would request all the reviewers to please just review and approve this PR.**

File: microsoft365defender_connector.py
Comment (alexa-phantom) [inline microsoft365defender_connector.py:228]: Can you please explain why this is the default app_name?

File: microsoft365defender_connector.py
Comment (alexa-phantom) [inline microsoft365defender_connector.py]: All occurrences of `phantom` needs to be updated to `SOAR`, including variable names

**Review by alexa-phantom — CHANGES_REQUESTED** [review]

File: microsoft365defender_connector.py
Comment (hsrivastava-crest) [reply microsoft365defender_connector.py:228]: Hi @alexa-phantom,
As per my understanding, if condition at line 227 can only become true when the app name in the app JSON does not contain any alphanumeric characters. 

As the reason for this, is _get_dir_name_from_app_name() method is getting called from _get_app_rest_url(). Here we are fetching the app_name variable from the app JSON (lines 620 - 623). But if any app name is not an alpha-numeric string then this condition will become true. Hence, for that case, only this condition has been kept for the edge case handling.

But if any reviewer knows any other reason apart from this then please let me know.

File: microsoft365defender_connector.py
Comment (hsrivastava-crest) [reply microsoft365defender_connector.py]: Hi @alexa-phantom,
We have followed the naming convention that is followed in the BaseConnector class. There also there is a method called "get_phantom_base_url()". Hence, we have kept the name of the variable as "phantom_base_url".

Comment (alexa-phantom) [reply microsoft365defender_connector.py]: Do you think we can change it to `Using SOAR base URL as` instead?

Comment (dhwanis-crest) [reply microsoft365defender_connector.py]: Updated

File: microsoft365defender_connector.py
Comment (rgil-splunk) [inline microsoft365defender_connector.py]: can we avoid hardcoding this value and make  this a variable/constant? maybe have a total wait time (105) and then get the loop end value by doing floor division of `total_time // sleep_time`

also, `range(0, N)` == `range(N)`

File: microsoft365defender_connector.py
Comment (rgil-splunk) [inline microsoft365defender_connector.py]: what does this hardcoded parameter mean? is it querying the top N matches? can we add a comment in the code explaining what it's doing?

**Review by rgil-splunk — APPROVED** [review]

File: microsoft365defender_connector.py
Comment (dhwanis-crest) [reply microsoft365defender_connector.py]: Resolved

**Review by alexa-phantom — APPROVED** [review]

**Pull request merged**
